src/js/gpio.js

```diff
@@ -47,47 +40,75 @@ GPIO.release = function() {
 };
 
 
-GPIO.pinmode = function(portpin, callback) {
-  var err = gpioctl.pinmode(portpin);
+function getDirectionCode(direction) {
+  if (direction == 'out') return gpioctl.OUT;
+  if (direction == 'in') return gpioctl.IN;
+  if (direction == 'none') return gpioctl.NONE;
+  return -1;
+}
+
+function getModeCode(mode) {
+  if (mode == 'pullup') return gpioctl.PULLUP;
+  if (mode == 'pulldn') return gpioctl.PULLDN;
+  if (mode == 'float') return gpioctl.FLOAT;
+  if (mode == 'pushpull') return gpioctl.PUSHPULL;
+  if (mode == 'opendrain') return gpioctl.OPENDRAIN;
+  if (mode == undefined || mode == 'none' || mode == '') return gpioctl.NONE;
+  return -1;
+}
 
-  if (util.isFunction(callback)) {
-    process.nextTick(function() {
-      callback(err);
-    });
+GPIO.setPin = function(pinNumber, direction, mode, callback) {
+  if (util.isFunction(mode)) {
+    callback = mode;
+    mode = '';
   }
-  else {
-    return err;
+  var dircode = getDirectionCode(direction);
+  if (dircode < 0) {
+    return gpioctl.ERR_INVALIDPARAM;
```

Comment:
According to GPIO API. we need to call callback function with corresponding `Error` object.
```
var err = new Error("Invalid parameter");
if (util.isFunction(callback)) {
  nextTick(function() {
    callback(err);
  });
} else {
  throw err;
}
```